File: src/main/java/me/xentany/xcore/CorePlugin.java
package me.xentany.xcore;

import me.xentany.xcore.util.AntiRelogUtils;
import org.bukkit.plugin.java.JavaPlugin;

public final class CorePlugin extends JavaPlugin {

  private static CorePlugin instance;

  @Override
  public void onEnable() {
    CorePlugin.instance = this;
    AntiRelogUtils.load();
  }
  
  @Override
  public void onDisable() {
  }
  
  public static CorePlugin getInstance() {
    return CorePlugin.instance;
  }
}

File: src/main/kotlin/me/xentany/xcore/CorePlugin.kt
package me.xentany.xcore

import me.xentany.xcore.util.hook.PlayerPointsProvider
import me.xentany.xcore.util.hook.VaultEconomyProvider
import me.xentany.xcore.util.hook.luckperms.LuckPermsProvider
import me.xentany.xcore.util.text.TextSerializer
import org.bukkit.plugin.java.JavaPlugin

internal class CorePlugin : JavaPlugin() {

  override fun onEnable() {
    LuckPermsProvider.hook(this)
    PlayerPointsProvider.hook(this)
    VaultEconomyProvider.hook(this)
  }

  override fun onDisable() {
    TextSerializer.clear()
  }
}